src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,365 @@
+                break;
+            }
+        }
+    }
+
+    private static void setTaskAsDone(String isCompleteString) {
+        if (isCompleteString.equals(NUMBER_DONE)) {
+            tasks.get(tasks.size() - 1).setDone();
+        }
+    }
+
+    private static void writeToFile(String taskInstance, String rawText, String additionalText, boolean isDone) throws IOException {
+        FileWriter fw = new FileWriter("data/duke.txt", true);
+        String additionalTextWithBorders = (additionalText.equals("") ? "" : " | " + additionalText);
+        String taskAsText = taskInstance + " | " + isDoneString(isDone) + " | " + rawText + additionalTextWithBorders + System.lineSeparator();
+        fw.write(taskAsText);
+        fw.close();
+    }
+
+    private static String isDoneString(boolean isDone) {
+        if (isDone) {
+            return "1";
+        } else {
+            return "0";
+        }
+    }
+
+    private static void showBottomMessage() {
+        System.out.println(DASHES);
+    }
+
+    private static void executeCommand(String userCommand) {
+        try {
+            final String[] commandTypeAndParams = splitUserCommand(userCommand);
+            final String commandType = commandTypeAndParams[0];
+            final String commandArgs = commandTypeAndParams[1];
+            switch (commandType) {
+            case COMMAND_LIST_WORD:
+                executeListTasks();
+                break;
+            case COMMAND_DELETE_WORD:
+                checkValidArguments(commandArgs);
+                executeDeleteTask(commandArgs);
+                break;
+            case COMMAND_COMPLETED_WORD:
+                checkValidArguments(commandArgs);
+                executeCompleteTask(commandArgs);
+                break;
+            case COMMAND_TODO_WORD:
+                checkValidArguments(commandArgs);
+                executeTodoTask(commandArgs);
+                break;
+            case COMMAND_DEADLINE_WORD:
+                checkValidArguments(commandArgs);
+                executeDeadlineTask(commandArgs);
+                break;
+            case COMMAND_EVENT_WORD:
+                checkValidArguments(commandArgs);
+                executeEventTask(commandArgs);
+                break;
+            case COMMAND_EXIT_WORD:
+                executeExitProgram();
+                break;
+            default:
+                throw new InvalidException();
+            }
+        } catch (InvalidException e) {
+            System.out.println("Sorry I didn't understand that :( Please try again");
+        } catch (EmptyTaskException e) {
+            System.out.println("Oops! Your task description cannot be empty D:");
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("Oops! The task index doesn't exist! Pls try again");
+        }
+    }
+
+    private static void executeDeleteTask(String taskIndexString) throws IndexOutOfBoundsException{
```

Comment:
Generally, good job in keeping most methods within the [30 LoC mark](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-long-methods). 

It would also help to have method descriptors to inform readers on what they are doing.